tests/common/runner.rs

```diff
@@ -62,15 +62,10 @@ pub fn test_execute_alloy<S: Storage + Send + Sync, C: PevmChain + Send + Sync +
     if must_match_block_header {
         let spec_id = chain.get_block_spec(&block.header).unwrap();
 
-        // We can only calculate the receipts root from Byzantium.
-        // Before EIP-658 (https://eips.ethereum.org/EIPS/eip-658), the
-        // receipt root is calculated with the post transaction state root,
-        // which we don't have in these tests.
-        if block.header.number >= 4370000 {
-            assert_eq!(
-                block.header.receipts_root,
-                chain.calculate_receipt_root(spec_id, &block.transactions, &tx_results)
-            );
+        match chain.calculate_receipt_root(spec_id, &block.transactions, &tx_results) {
+            Ok(None) => {}
+            Ok(Some(receipt_root)) => assert_eq!(block.header.receipts_root, receipt_root),
+            Err(err) => panic!("{:?}", err),
```

Comment:
This is stricter than needed since we cannot guarantee to get a receipt root out of all tests. We can just do this as "if we can get a receipt root here then assert it". Otherwise, this case (receipt root calculation correctness) should be included in dedicated unit tests of each chain.

Reply:
Can you check my last commit? Is this solution ok?

Reply:
Not perfect but definitely good enough for now, let's merge.

Reply:
wait, let me resolve all the discussions first. Some are not reflected in the code.

Reply:
> Not perfect

Yeah, I understand that this solution does not allow other chain to customize the `Custom` arm. Using `String` is not perfect.